core/src/main/java/org/wso2/testgrid/core/command/GenerateEscalationEmailCommand.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.core.command;
+
+import org.kohsuke.args4j.Option;
+import org.kohsuke.args4j.spi.StringArrayOptionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.exception.CommandExecutionException;
+import org.wso2.testgrid.reporting.EscalationEmailGenerator;
+import org.wso2.testgrid.reporting.ReportingException;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is responsible for generating the escalation email for all products.
+ */
+public class GenerateEscalationEmailCommand implements Command {
+
+    @Option(name = "--workspace",
+            usage = "Product workspace",
+            aliases = {"-w"},
+            required = true)
```

Comment:
We can make workspace optional?

Reply:
If we make this optional the pipeline could break since Email reports and Graphs could be generate else where.